File: src/BankAccount.java
public class BankAccount{
    private long accountNumber;
    private double amount;
    private char type;

    public BankAccount(long accountNumber, char type) {
        this.accountNumber = accountNumber;
        this.type = type;
    }

    public long getAccountNumber() {
        return accountNumber;
    }

    public char getType() {
        return type;
    }

    public double getAmount() {
        return amount;
    }

    public void agregarDinero(double amount) {
        if (type == 'A') agregarDineroA(amount);
        else if (type == 'B') agregarDineroB(amount);
        else {
            this.amount += amount;
            System.out.printf("Se han agregado: %f a su cuenta",amount);
        }
    }

    private void agregarDineroA(double amount) {
        if (this.amount + amount <= 50000) {
            this.amount += amount;
            System.out.printf("Se han agregado: %f a su cuenta",amount);
        } else {
            System.out.println("No puede tener más de 50,000 cuenta A");
        }
    }

    private void agregarDineroB(double amount) {
        if (this.amount + amount <= 100000) {
            this.amount += amount;
            System.out.printf("Se han agregado: %f a su cuenta",amount);
        } else {
            System.out.println("No puede tener más de 100,000 cuenta B");
        }
    }
    public void retirarDinero(double retiro){
        switch(type){
            case 'A':
                retirarDineroA(retiro);
                System.out.printf("Se han retirado: %f de su cuenta, su nuevo saldo es %d",retiro,amount);
                break;

            case 'B':
                retirarDineroB(retiro);
                System.out.printf("Se han retirado: %f de su cuenta, su nuevo saldo es %d",retiro,amount);
                break;

            case 'C':
                retirarDineroC(retiro);
                System.out.printf("Se han retirado: %f de su cuenta, su nuevo saldo es %d",retiro,amount);
                break;
        }
    }
    private void retirarDineroA(double retiro){
        if(amount < 1000){
            System.out.println("\nSaldo insuficiente para retirar");
        }
        else if(amount < retiro){
            System.out.println("\nEl dinero a retirar es mayor al saldo disponible");
        }
        else{
            amount -= retiro;
        }
    }

    private void retirarDineroB(double retiro){
        if(amount < 5000){
            System.out.println("\nSaldo insuficiente para retirar");
        }
        else if(amount < retiro){
            System.out.println("\nEl dinero a retirar es mayor al saldo disponible");
        }
        else{
            amount -= retiro;
        }
    }

    private void retirarDineroC(double retiro){
        if(amount < 10000){
            System.out.println("\nSaldo insuficiente para retirar");
        }
        else if(amount < retiro){
            System.out.println("\nEl dinero a retirar es mayor al saldo disponible");
        }
        else{
            amount -= retiro;
        }
    }
}

File: src/BankAccount.py
class BanckAccount:
    __amount=0
    def __init__(self,account_number,type):
        self.__account_number=account_number
        self.__type=type
    
        
    def get_account_number(self):
        return self.__account_number 

    def get_type(self):
        return self.__type

    def get_amount(self):
        return self.__amount

    def agregar_dinero(self,amount):
        if self.__type=='A':
            self.agregar_dinero_a(amount)
        elif self.__type=='B':
            self.agregar_dinero_b(amount)
        else:
            self.__amount += amount
            print("Se ha agregado",amount,"a su cuenta")

    def agregar_dinero_a(self,amount):
        if self.__amount+amount<=50000:
            self.__amount+=amount
            print("Se ha agregado",amount,"a su cuenta")
        else:
            print("No se puede tener mas de 50,000 en cuenta A")

    def agregar_dinero_b(self,amount):
        if self.__amount+amount<=100000:
            self.__amount+=amount
            print("Se ha agregado",amount,"a su cuenta")
        else:
            print("No se puede tener mas de 100,000 en cuenta B")
        
    def retirar_dinero(self,amount):
        if self.__type=='A':
            self.retirar_dinero_a(amount)
            print("Se ha retirado",amount,"de su cuenta, su nuevo saldo es:",self.__amount)
        elif self.__type=='B':
            self.retirar_dinero_b(amount)
            print("Se ha retirado",amount,"de su cuenta, su nuevo saldo es:",self.__amount)
        elif self.__type=='C':
            self.retirar_dinero_c(amount)
            print("Se ha retirado",amount,"de su cuenta, su nuevo saldo es:",self.__amount)
             
    def retirar_dinero_a(self,retiro):
        if self.__amount<1000:
            print("Saldo insuficiente para retirar")
        elif self.__amount<retiro:
            print("El dinero a retirar es mayor al saldo disponible")
        else:
            self.__amount-=retiro
    
    def retirar_dinero_b(self,retiro):
        if self.__amount<5000:
            print("Saldo insuficiente para retirar")
        elif self.__amount<retiro:
            print("El dinero a retirar es mayor al saldo disponible")
        else:
            self.__amount-=retiro
    
    def retirar_dinero_c(self,retiro):
        if self.__amount<10000:
            print("Saldo insuficiente para retirar")
        elif self.__amount<retiro:
            print("El dinero a retirar es mayor al saldo disponible")
        else:
            self.__amount-=retiro